asapdiscovery-data/asapdiscovery/data/schema_v2/target.py

```diff
@@ -91,34 +93,30 @@ def _validate_at_least_one_id(cls, v):
 
     @classmethod
     def from_pdb(
-        cls, pdb_file: Union[str, Path], target_name: Optional[str] = None, **kwargs
+        cls, pdb_file: Union[str, Path], target_chains=[], ligand_chain="", **kwargs
     ) -> "Target":
+        kwargs.pop("data", None)
         # directly read in data
-        pdb_str = read_file_directly(pdb_file)
-        return cls(data=pdb_str, target_name=target_name, **kwargs)
+        # First load full complex molecule
+        complex_mol = load_openeye_pdb(pdb_file)
 
-    @classmethod
-    def from_pdb_via_openeye(
-        cls, pdb_file: Union[str, Path], target_name: Optional[str] = None, **kwargs
-    ) -> "Target":
-        # directly read in data
-        pdb_str = read_file_directly(pdb_file)
-        # NOTE: tradeof between speed and consistency with `from_pdb` method lines below will make sure that the pdb string is
-        # consistent between a load and dump by roundtripping through and openeye mol but will slow down the process significantly.
-        mol = pdb_string_to_oemol(pdb_str)
-        pdb_str = oemol_to_pdb_string(mol)
-        return cls(data=pdb_str, target_name=target_name, **kwargs)
+        # Split molecule into parts using given chains
+        mol_filter = MoleculeFilter(
+            protein_chains=target_chains, ligand_chain=ligand_chain
+        )
+        split_dict = split_openeye_mol(complex_mol, mol_filter)
+
+        return cls.from_oemol(split_dict["prot"], **kwargs)
 
     def to_pdb(self, filename: Union[str, Path]) -> None:
         # directly write out data
         write_file_directly(filename, self.data)
 
     @classmethod
-    def from_oemol(
-        cls, mol: oechem.OEMol, target_name: Optional[str] = None, **kwargs
-    ) -> "Target":
+    def from_oemol(cls, mol: oechem.OEMol, **kwargs) -> "Target":
```

Comment:
is there a reason that we're getting rid of the `target_name` arg other than that it's not really necessary to have since it can just get lumped in `kwargs`?

Reply:
Yeah its messing with dumping via kwargs as can sometimes get duplicate kwarg as you say.